primitives/substrate/src/lib.rs

```diff
@@ -0,0 +1,108 @@
+// Copyright 2019-2020 Parity Technologies (UK) Ltd.
+// This file is part of Parity Bridges Common.
+
+// Parity Bridges Common is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity Bridges Common is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity Bridges Common.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Primitives for the Substrate light client (a.k.a bridge) pallet.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use core::default::Default;
+use parity_scale_codec::{Decode, Encode};
+#[cfg(feature = "std")]
+use serde::{Deserialize, Serialize};
+use sp_finality_grandpa::{AuthorityList, SetId};
+use sp_runtime::traits::Header as HeaderT;
+use sp_runtime::RuntimeDebug;
+
+/// A Grandpa Authority List and ID.
+///
+/// The list contains the authorities for the current round, while the
+/// set id is a monotonic identifier of the current authority set.
+#[derive(Default, Encode, Decode, RuntimeDebug, PartialEq, Clone)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub struct AuthoritySet {
+	pub authorities: AuthorityList,
+	pub set_id: SetId,
+}
+
+impl AuthoritySet {
+	pub fn new(authorities: AuthorityList, set_id: SetId) -> Self {
+		Self { authorities, set_id }
+	}
+}
+
+/// Keeps track of when the next Grandpa authority set change will occur.
+///
+/// The authority set stored here is expected to be enacted at a block height
+/// of N, assuming we get a valid justification for the header at N.
+#[derive(Default, Encode, Decode, RuntimeDebug, PartialEq, Clone)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub struct ScheduledChange<N> {
+	pub authority_set: AuthoritySet,
+	pub height: N,
+}
+
+impl<N> ScheduledChange<N> {
+	pub fn new(authority_set: AuthoritySet, height: N) -> Self {
```

Comment:
Is it really needed? Both fields are public and it's just more readable if you initialize them explicitly.